### ISSUE TITLE
Use UTF-8 decode without BOM for multipart/form-data

### ISSUE THREAD

**Issue by CYBAI** [servo/servo#24648]
whatwg/fetch#915
web-platform-tests/wpt#17589 (not merged yet)

@jdm I think following code is the related function 👀?

https://github.com/servo/servo/blob/f626355b67fc007d5961c446d5a4ddbcc3785698/components/script/body.rs#L173-L200

**Comment by jdm** [comment]
You are correct, and it's also related to https://github.com/servo/rust-url/blob/7d2c9d6ceb3307a3fad4c4bcb7f5b07b5e24c755/src/form_urlencoded.rs#L21-L58. It's not entirely clear to me what needs to be done.
